crates/bevy_input/src/gamepad.rs

```diff
@@ -1240,6 +1240,83 @@ const ALL_AXIS_TYPES: [GamepadAxisType; 6] = [
     GamepadAxisType::RightZ,
 ];
 
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct GamepadRumbleIntensity {
+    /// The rumble intensity of the strong gamepad motor
+    ///
+    /// Ranges from 0.0 to 1.0
+    pub strong: f32,
+    /// The rumble intensity of the weak gamepad motor
+    ///
+    /// Ranges from 0.0 to 1.0
+    pub weak: f32,
+}
+
+impl GamepadRumbleIntensity {
+    /// Rumble both gamepad motors at maximum intensity
+    pub const MAX: Self = GamepadRumbleIntensity {
+        strong: 1.0,
+        weak: 1.0,
+    };
+
+    /// Don't rumble at all, only makes sense when `additive` is `false`
+    pub const ZERO: Self = GamepadRumbleIntensity {
+        strong: 0.0,
+        weak: 0.0,
+    };
+}
+
+/// Request that `gamepad` should rumble with `intensity` for `duration_seconds`
+///
+/// # Notes
+///
+/// * Does nothing if the `gamepad` does not support rumble
+/// * If a new `GamepadRumbleRequest` is sent while another one is still executing, it
+///   replaces the old one.
+///
+/// # Example
+///
+/// ```
+/// # use bevy_gilrs::{GamepadRumbleRequest, GamepadRumbleIntensity};
+/// # use bevy_input::gamepad::Gamepad;
+/// # use bevy_app::EventWriter;
+/// fn rumble_pad_system(mut rumble_requests: EventWriter<GamepadRumbleRequest>) {
+///     let request = GamepadRumbleRequest {
+///         intensity: GamepadRumbleIntensity::Strong,
+///         duration_seconds: 10.0,
+///         gamepad: Gamepad(0),
+///         additive: true,
+///     };
+///     rumble_requests.send(request);
+/// }
+/// ```
+#[derive(Clone)]
+pub struct GamepadRumbleRequest {
```

Comment:
Would it be better to make the request an enum instead of having an `additive` field? Something like this?

```rust
#[derive(Clone)]
pub enum GamepadRumbleRequest {
    /// Add a rumble to the given gamepad.
    ///
    /// To replace a rumble, Send a `Stop` event first.
    Add {
        gamepad: Gamepad,
        duration_seconds: f32,
        intensity: GamepadRumbleIntensity,
    },
    Stop {
        gamepad: Gamepad,
    }
}
```

Reply:
yes I think so.

Reply:
Definitely agree on the enum form.